dice-game-ui-angular/src/app/services/local-game-storage-service.ts

```diff
@@ -1,33 +1,70 @@
 import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
-  createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
-  }
+  private readonly savedGamesKey = 'saved-games';
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    const allSavedGames = this.getAllSavedGames();
+    if (!guid && allSavedGames.length === 0) {
+      throw new Error('This game was not created in the store.');
+    }
+
+    let savedGame: SavedGameEngine = allSavedGames.find(
+      (sg) => sg.guid === (guid ? guid.toString() : Guid.EMPTY)
+    );
+    // if saving previously saved game
+    if (guid) {
+      if (!savedGame) {
+        throw new Error('This game was not created in the store.');
+      }
+      savedGame.gameEngine = gameEngine;
+
+      // if saving a new game
+    } else {
+      if (savedGame) {
+        throw new Error('This game was previously created in the store.');
+      }
 
-  saveGame(gameEngine: GameEngine, guid: Guid): Guid {
-    if (!localStorage.getItem(guid.toString())) {
-      throw new Error('The saved game was not found in the store');
+      savedGame = new SavedGameEngine(gameEngine);
+      allSavedGames.push(savedGame);
+      guid = Guid.parse(savedGame.guid);
     }
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
+
+    localStorage.setItem(this.savedGamesKey, JSON.stringify(allSavedGames));
     return guid;
   }
 
   restoreGame(guid: Guid): GameEngine {
-    const fieldSize: number = JSON.parse(localStorage.getItem(guid.toString()));
-    if (!fieldSize) {
+    const allSavedGames = this.getAllSavedGames();
+    if (allSavedGames.length === 0) {
```

Comment:
This all can be done easier.
```ts
const curSavedGame = allSavedGames.find(sg => sg.guid === guid.toString());
return curSavedGame
  ? new SimpleGameEngine(curSavedGame.fieldSize.width, curSavedGame.fieldSize.height)
  : null;
```

Reply:
Changed

Reply:
Maybe it is better to remove this check and add check that saved game with such guid exists instead of returning null.